### PR TITLE
Plugin upgrades

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- Adds a button "Check for upgrades" that checks each plugin if it's the latest github commit / gitlab commit / npm version. If not, you can click to update.

- If there's an error connecting to the backend (e.g. github api key limit exceeded, happens often), we show an error:
image

- If a plugin is enabled, we update it, and afterwards there's a new required field, we directly open the config editor:

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

**Comment by timgl** [comment]
I think it'd be cool to just run this whenever the page loads rather than on the button click. Much higher chance people will see & actually update

File: frontend/src/custom.d.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+declare module '*.svg' {
+    const content: any
+    export default content
+}
```
Comment (Twixes) [inline frontend/src/custom.d.ts:4]: What does this do?

Comment (Twixes) [inline frontend/src/custom.d.ts:4]: What does this do?

**Review by Twixes — COMMENTED** [review]
100% should run automatically (though not more often than x amount of time).

File: frontend/src/custom.d.ts (hunk 1 above)
Comment (mariusandra) [reply frontend/src/custom.d.ts:4]: Added as a comment into the file:
```js
// This fixes TS errors when importing a .svg file
```

**Comment by mariusandra** [comment]
The plugins will now check for updates when the page opens. More specifically, I set it to cache the "latest_tag" for each plugin for 30min (with postgres) and it'll only check on page load if the status isn't stale. Clicking on the "check for updates" button checks all plugins.

Currently clicking "check for all updates" tends to exhaust the github api limit quite quickly. This will be fixed when #3070 lands.

**Review by Twixes — APPROVED** [review]
Nice

**Comment by mariusandra** [comment]
I think that's a good thing...? See also #3082

Thanks for the review!

**Pull request merged**